crypto/ec/curve448/arch_32/arch_intrinsics.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017-2018 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright 2016 Cryptography Research, Inc.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ *
+ * Originally written by Mike Hamburg
+ */
+
+#ifndef __ARCH_ARCH_32_ARCH_INTRINSICS_H__
+# define __ARCH_ARCH_32_ARCH_INTRINSICS_H__
+
+# define ARCH_WORD_BITS 32
+
+static ossl_inline uint32_t word_is_zero(uint32_t a)
+{
+    /* let's hope the compiler isn't clever enough to optimize this. */
+    return (((uint64_t)a) - 1) >> 32;
```

Comment:
I can't find the previous comment about this anymore, probably because it was on a file that's now deleted.

It's true that we currently don't have a version that takes a uint32_t (and uint64_t), but we've already added an uint64_t of the msb function, it shouldn't be that hard to add a few more functions.

Reply:
Now using our own standard constant time code for everything constant time related.